server/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -908,59 +908,21 @@ public TranslogDeletionPolicy getDeletionPolicy() {
         return deletionPolicy;
     }
 
-    public static class Location implements Comparable<Location> {
+    public record Location(long generation, long translogLocation, int size) implements Comparable<Location> {
 
         public static Location EMPTY = new Location(0, 0, 0);
 
-        public final long generation;
-        public final long translogLocation;
-        public final int size;
-
-        public Location(long generation, long translogLocation, int size) {
-            this.generation = generation;
-            this.translogLocation = translogLocation;
-            this.size = size;
-        }
-
         @Override
         public String toString() {
             return "[generation: " + generation + ", location: " + translogLocation + ", size: " + size + "]";
         }
 
         @Override
         public int compareTo(Location o) {
-            if (generation == o.generation) {
-                return Long.compare(translogLocation, o.translogLocation);
-            }
-            return Long.compare(generation, o.generation);
-        }
-
-        @Override
-        public boolean equals(Object o) {
-            if (this == o) {
-                return true;
-            }
-            if (o == null || getClass() != o.getClass()) {
-                return false;
-            }
-
-            Location location = (Location) o;
-
-            if (generation != location.generation) {
-                return false;
+            int result = Long.compare(generation, o.generation);
+            if (result == 0) {
+                result = Long.compare(translogLocation, o.translogLocation);
             }
-            if (translogLocation != location.translogLocation) {
-                return false;
-            }
-            return size == location.size;
-
-        }
-
-        @Override
-        public int hashCode() {
```

Comment:
We never store this, right? And thus we can live with whatever implementation the JVM provides for records hashCode. Or should we re-use the same hashCode to be safe?

Reply:
The tests pass with `hashCode` poisoned, so it looks like it isn't used at all